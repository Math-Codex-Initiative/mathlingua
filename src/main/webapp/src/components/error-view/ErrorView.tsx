import styles from './ErrorView.module.css';

interface ErrorViewProps {
  message: string;
}

export function ErrorView(props: ErrorViewProps) {
  return (
    <div className={styles.errorMessage}>
      <span className={styles.errorLabel}>ERROR:</span> {props.message}
    </div>
  );
}
