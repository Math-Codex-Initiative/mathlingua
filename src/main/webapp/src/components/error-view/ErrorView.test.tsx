import { render, screen } from '@testing-library/react';
import { ErrorView } from './ErrorView';

test('it renders', () => {
  render(<ErrorView message="some custom error" />);
  const el = screen.getByText('some custom error');
  expect(el).toBeInTheDocument();
  expect(el).toMatchSnapshot();
});
