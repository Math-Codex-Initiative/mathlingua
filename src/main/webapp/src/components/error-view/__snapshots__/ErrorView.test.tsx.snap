// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
<div
  class="errorMessage"
>
  <span
    class="errorLabel"
  >
    ERROR:
  </span>
   
  some custom error
</div>
`;
